### DISCUSSION TITLE
CMF - Academy Question

### DISCUSSION THREAD

**Discussion by amasotti** [pimcore/pimcore#9426] [Q&A / Support] [answered]
Hi, 

I am really confused about one question in Academy - Senior Dev Course. In the chapter about CMF there is a question (nr. 2), that goes as follows:

*What data is managed  with Pimcore Data Objects*?
- Tracking Infos
-  Activities
- Customer
- Customer segments

I've watched the videos a couple of times and read the docs. Now this is my reasoning:
- `Customers` are objects for sure (this was repeated many times and can definitely be managed as any other DataObject)
- `Customer segments` are - as far as I understand - also Data objects: we can manage them as normal objects, assign them data types and attributes as with any other object. In the docs you can read: 
    > CustomerSegments and CustomerSegmentGroups are regular Pimcore objects that are assigned via relations to customer objects.

-  `Activities` : although not explicitly stated in the videos, the docs have [a section about treating Activities as DataObjects](https://pimcore.com/docs/customer-management-framework/current/Activities/index.html).
-  `Tracking Information`: I am not quite sure what do you mean here. VisitorInformation (id, session etc...) or ActivityLink tracking? 

Anyway keeping Customer and Customer segments as answers, I have tried all other possible combinations with or without `activities` and `tracking info` (one of them and both together) but I am stuck with a "exam failed".

What I am missing here? What am I understanding wrong?

**Comment by fashxp** [comment, marked as answer]
Hey, 
see following a few comments: 

- Customers are objects for sure (this was repeated many times and can definitely be managed as any other DataObject)  👍

- Customer segments are - as far as I understand - also Data objects: we can manage them as normal objects, assign them data types and attributes as with any other object. In the docs you can read ... 👍

- Activities : although not explicitly stated in the videos, the docs have a section about treating Activities as DataObjects.

Thats a good point. But still, the docs also says: 
> Although these activities have their own data persistence they are additionally stored in the generic ActivityStore. The idea is that the ActivityStore saves all activities in a standardized way and represents a history of all customer activities.

Maybe we should rephrase the question to _What data is **primarily** managed with Pimcore Data Objects?_ 
So activities are a wrong answer here. 👎

- Tracking Information: I am not quite sure what do you mean here. VisitorInformation (id, session etc...) or ActivityLink tracking?

Also for link activity tracking, only configuration is managed with data objects. The actual tracking information is never stored in data objects (as this would be a huge overhead). 
So tracking information is also a wrong answer here. 👎

**Reply by amasotti** [reply]
Thanks!

I agree that *What data is **primarily**...* would be a much clearer phrasing for this question.
My problem was indeed with Tracking Info, now I understand what you mean and how this works.
